src/core/dbus-cgroup.c

```diff
@@ -396,6 +398,7 @@ const sd_bus_vtable bus_cgroup_vtable[] = {
         SD_BUS_PROPERTY("ManagedOOMSwap", "s", property_get_managed_oom_mode, offsetof(CGroupContext, moom_swap), 0),
         SD_BUS_PROPERTY("ManagedOOMMemoryPressure", "s", property_get_managed_oom_mode, offsetof(CGroupContext, moom_mem_pressure), 0),
         SD_BUS_PROPERTY("ManagedOOMMemoryPressureLimitPercent", "s", bus_property_get_percent, offsetof(CGroupContext, moom_mem_pressure_limit), 0),
+        SD_BUS_PROPERTY("AllowBindPorts", "as", NULL, offsetof(CGroupContext, allow_bind_ports), 0),
```

Comment:
I think in general there's a preference for passing around structured data, rather than strings to parse - maybe this could be an int, with -1 to indicate "none"?

Reply:
@bluca IMO `-1` semantics is confusing while passing "none" would be consistent with service unit config file.
Also there are IPAddress{Allow|Deny} properties with special values such as "any" and "localhost", I followed that pattern.

Reply:
In general, with the DBUS interface we don't really aim for 1:1 mapping with unit config, but for well-structured data passing. And IPAddress is passed around as structured data (array of bytes + unsigned for mask), not as strings:

```
        SD_BUS_PROPERTY("IPAddressAllow", "a(iayu)", property_get_ip_address_access, offsetof(CGroupContext, ip_address_allow), 0),
        SD_BUS_PROPERTY("IPAddressDeny", "a(iayu)", property_get_ip_address_access, offsetof(CGroupContext, ip_address_deny), 0),
```

any/localhost/etc are encoded as special values - eg, 0.0.0.0/0

Reply:
@bluca 

Is integer (i.e.`t` dbus type specifier) forward compatible?
It keep the initial program simple and to reduce the complexity of this PR, `AllowBindPorts=` accepts a single integer or `none`. 
Still I'd like to add range support as follow-up, e.g. `AllowBindPorts=5555:7777` which will require a string.

Reply:
It can be two integers. You can immediately add a second one, but ignore it for now - that way when it's ready you can use it for ranges.

Reply:
yes, please use the correct type here, not strings. dbus is all about type-safety, and strings are about "everything goes", hence break tht design goal of dbus.

now, dbus isn't good at having nice ways to extend things in a compatible way. it usually pays of to think in advance how thing are going to be extended, and then use the appropriate types for that. and if that doesn't work out and one has to change type the usual approach is to introduce a new property that has the new type, and both the old and the new continue to be exposed, showing the same data as much as that's possible

Reply:
btw what about ipv4 vs. ipv6? here? eventually people probably want to list ports per protocol...

maybe encode "none" as single entry of value "0" (which is after all not a valid port, i.e. if the only allowed port is an invalid one i think the meaning is clear)

Reply:
@poettering 

Does `<protocol>:<port>` format, e.g. `IPv6:5555` look good? This may be extended to `<protocol>:<port_min>-<port_max>`, e.g.`IPv6:5555-7777` as port range support is added.

Reply:
use an array of structures of three integers: the address family to use (AF_INET or AF_INET6), and the port start nr, plus number of ports. the dbus type system is easily flexible enough to express that.

Reply:
@poettering @bluca 

Hey, dbus is switched to (address_family, nr_ports, port_min) three tuple.